Device/Device.cs

```diff
@@ -94,9 +106,9 @@ private async void Disconnected(MqttClientDisconnectedEventArgs e, IMqttClientOp
                 Console.WriteLine("Trying to reconnect");
                 await mqttClient.ConnectAsync(options, CancellationToken.None);
             }
-            catch
+            catch (Exception ex)
             {
-                Console.WriteLine("### RECONNECTING FAILED ###");
+                Console.WriteLine("### RECONNECTING FAILED ###" + ex.Message);

```

Comment:
Nitpick: Maybe add a new write line instead of appending:
`                
Console.WriteLine("With exception: " + ex.Message);
`